test/test-helpers.js

```diff
@@ -30,7 +30,9 @@ function timePasses(ms) {
 const flushCb = window.flush;
 window.flush = () => new Promise((resolve) => {
   flushCb(() => {
-    resolve();
+    window.requestIdleCallback(() => {
```

Comment:
`requestIdleCallback` is not supported on Edge and Safari. We need a fallback mechanism here.